anki: more no-op exp cleanup

bhipple opened the pull request:
Continued progress towards #89745; just simplifying a bit.


<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!
List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://hydra.nixos.org/job/nixpkgs/trunk/manual/latest/download/1/nixpkgs/manual.html#chap-reviewing-contributions
-->

###### Motivation for this change


###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [X] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [X] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [X] Ensured that relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

Profpatsch commented on pkgs/games/anki/default.nix:
```diff
@@ -31,7 +31,7 @@
 let
   # when updating, also update rev-manual to a recent version of
-  # https://github.com/dae/ankidocs
+  # https://github.com/ankitects/ankidocs
```
dead link

bhipple replied:
Oops, it's https://github.com/ankitects/anki-docs now, with a dash on the RHS.

Profpatsch commented on pkgs/games/anki/default.nix:
```diff
@@ -42,7 +42,7 @@ let
     inherit version;
     src = fetchFromGitHub {
-      owner = "dae";
+      owner = "ankitects";
```
That’s not a no-op holy <s>f</s> heck.

I do hope that slipped into this commit by mistake.

Ekleog replied:
This being said, it looks like ankisrs.net now points to the ankitects repo, so IMO this is not a problem (even though it doesn't match the commit name). (Also, the build will probably not be affected, as the reference and hash stay the same)

bhipple replied:
https://github.com/dae/ankidocs is a URL redirect to https://github.com/ankitects/anki-docs now, so it is a no-op.  I can change the `sha256` to garbage and nix will re-fetch and tell me the same hash.  Am I missing something?

Profpatsch replied:
Yeah, please put it in a different commit explaining that.

bhipple replied:
Done

Profpatsch commented on pkgs/games/anki/default.nix:
```diff
@@ -112,16 +112,10 @@ buildPythonApplication rec {
   postPatch = ''
-    # Remove unused starter. We'll create our own, minimalistic,
-    # starter.
-    # rm anki/anki
```
Is this not relevant anymore?

bhipple replied:
It was commented out back in 2018 and we seem to have been fine without it, so I'd say so:
https://github.com/NixOS/nixpkgs/commit/a30e30432da95d6eae8713b57ca5708d892a3263

bhipple committed:
anki: update download url

The original repo has been transferred on github, so the URL is a redirect to here.

bhipple committed:
anki: expression cleanup
